### PR TITLE
add Junior Developer Rules doc

### PR DESCRIPTION
**Pull request by suspect15**
Adds page of all rules governing the Junior Developer Program.

This doc is intended to be a one-stop shop for people to understand the requirements for participation in the Junior Developer Program including how to join, how to remain in good standing and the requirements for requesting a code review.  Previously, rules were scattered in various locations such as the welcome message, #jr-devs discord pins, #developer-news announcements, developer code of conduct docs and forum posts.

This doc has been reviewed and approved by the Code Reviewer team.  Each new rule was separately discussed and voted on.

### CODE REVIEW

File: docs/developer-docs/jr-dev-rules.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+title: Junior Developer Program Rules
+description: Consolidates scattered junior developer program rules, adds clarification and introduces some new rules
```
Comment (wescopeland) [inline docs/developer-docs/jr-dev-rules.md]: This is what will appear in the Discord embed and on external search engines. Recommend something like:

```md
description: Comprehensive guidelines for the Junior Developer program, including rules, guidelines, and clarifications.
```

Comment (suspect15) [reply docs/developer-docs/jr-dev-rules.md]: Updated as recommended.  I appreciate the SEO tips.

File: docs/developer-docs/jr-dev-rules.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+## Overview
+
+The purpose of the Junior Developer Program is to assist interested people with learning how to create and maintain high quality achievement sets for RetroAchievements. Junior Developers are required to understand and abide by the [Developer Code of Conduct](https://docs.retroachievements.org/guidelines/users/code-of-conduct.html) except where specifically addressed by this document.
+
+### Entry Requirements
+
+To join the Junior Developer Program, the following requirements must be met:
+- RetroAchievements account must not be banned or suspended from achievement development or involuntarily untracked from competitive leaderboards
+- Have a Discord account verified on the [RetroAchievements Discord server](https://discord.gg/dq2E4hE)
+- Request the Junior Developer role in #role-request channel in the General section of the RA discord server
```
Comment (wescopeland) [inline docs/developer-docs/jr-dev-rules.md]:
```suggestion
- Request the Junior Developer role in #role-request channel in the General section of the RA Discord server
```

File: docs/developer-docs/jr-dev-rules.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  - Include RA profile link
```
Comment (wescopeland) [inline docs/developer-docs/jr-dev-rules.md]:
```suggestion
  - Include your RA profile link
```

File: docs/developer-docs/jr-dev-rules.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+- May not submit sets that do not have original achievement badges
+- May not submit sets that do not have Rich Presence
+- May not request a code review until at least 7 days after claiming a set
+- Will describe in the Code Review request the self testing method used in order to validate set stability
+
+**Sets that fail to meet Code Review request requirements will be refused entry into the Code Review backlog until remediated**
+
+### Removal From Junior Developer Program
+
+Participation in the Junior Developer program is not an entitlement or right. Junior Developers are expected to take the program seriously, put forth significant effort, seek assistance when unable to figure something out, strive to make the highest quality set possible and maturely accept and apply Code Reviewer feedback. Junior developers may be temporarily or permanently removed from the program for the following reasons:
```
Comment (wescopeland) [inline docs/developer-docs/jr-dev-rules.md]: Minor nit: I notice "Program" is capitalized in some contexts and lowercased in others. No opinion on which one is correct, just want to point it out.

**Review by wescopeland — COMMENTED** [review]
Looks good to me - just a few minor nits here and there after proofreading

File: docs/developer-docs/jr-dev-rules.md (hunk 2 above)
Comment (suspect15) [reply docs/developer-docs/jr-dev-rules.md]: Updated as recommended.

File: docs/developer-docs/jr-dev-rules.md (hunk 4 above)
Comment (suspect15) [reply docs/developer-docs/jr-dev-rules.md]: Updated as recommended.

File: docs/developer-docs/jr-dev-rules.md (hunk 3 above)
Comment (suspect15) [reply docs/developer-docs/jr-dev-rules.md]: Updated as recommended.

**Review by wescopeland — APPROVED** [review]

**Pull request merged**